streams/src/test/java/org/apache/kafka/streams/perf/SimpleBenchmark.java

```diff
@@ -348,8 +525,155 @@ public void close() {
         return new KafkaStreams(builder, props);
     }
 
+    private KafkaStreams createKafkaStreamsKStreamKTableJoin(String kStreamTopic, String kTableTopic, final long numRecordsExpected, File stateDir, String kafka, String zookeeper, final CountDownLatch latch) {
+        Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "simple-benchmark-kstream-ktable-join");
+        props.put(StreamsConfig.STATE_DIR_CONFIG, stateDir.toString());
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
+        props.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, zookeeper);
+        props.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.Long().getClass());
+        props.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.ByteArray().getClass());
+        final KStreamBuilder builder = new KStreamBuilder();
+
+        final KStream<Long, byte[]> input1 = builder.stream(kStreamTopic);
+        final KTable<Long, byte[]> input2 = builder.table(kTableTopic, kTableTopic + "-store");
+
+        final Long[] numRecordsProcessed = new Long[1];
+        numRecordsProcessed[0] = 0L;
+        ForeachAction<Long, byte[]> action =
+            new ForeachAction<Long, byte[]>() {
+                @Override
+                public void apply(Long key, byte[] value) {
+                    numRecordsProcessed[0]++;
+                    if (numRecordsExpected == numRecordsProcessed[0]) {
+                        latch.countDown();
```

Comment:
Why not just initialize the `CountDownLatch` with 'numRecordsExpected` and always call `latch.countDown()`? removes the need for the `Long[]` and the `if(..)`